lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -263,6 +272,15 @@ defmodule Thrift.Generator.StructBinaryProtocol do
   end
   defp map_key_deserializer(:double, key_name, value_name, _file_group) do
     quote do
+      defp unquote(key_name)(<<0::1, 2047::11, 0::52, rest::binary>>, stack) do
+        unquote(value_name)(rest, :inf, stack)
+      end
+      defp unquote(key_name)(<<1::1, 2047::11, 0::52, rest::binary>>, stack) do
+        unquote(value_name)(rest, :"-inf", stack)
+      end
+      defp unquote(key_name)(<<_::1, 2047::11, _::52, rest::binary>>, stack) do
+        unquote(value_name)(rest, :NaN, stack)
```

Comment:
We could have multiple NaNs disappear in this map as many possible keys are going to a single key, losing the values of all but one key. I think we should consider raising an error if can not define a one-to-one mapping.

Would we want to keep the sign and fraction information so that we can do loseless passthrough? That would mean %NaN{sign, fraction} or similar?

Reply:
I guess having a %NaN{sign, fraction} type makes sense. I'll edit this PR with such a type.